src/Types/Animations/BubblePop.tsx

```diff
@@ -1,98 +1,209 @@
 import 'p5collide';
-import { findDOMNode } from 'react-dom';
+import { newFigure } from './Animation';
 
 export class BubblePop extends Animation {
 
-    static burst(fig, sketchData) {
+    static NUM_FIGURES = 200;
+    static CACHE_LIMIT = 100;
+    static counter = 0;
 
-        if (fig.timer <= 25) {
-            let index = sketchData.figs.indexOf(fig);
-            sketchData.figs.splice(index, 1);
-            fig.pop.play();
+    static colorSettingsCache;
+    static circleCache = [];
+    static squareCache = [];
+    static triangleCache = [];
+
+    static burst_cache(fig, sketchData, p) {
+
+        let index = sketchData.figs.indexOf(fig);
+
+        let type : string = fig.getShapeDescriptor().type;
+        let dim : number = fig.dim;
+        let xpos : number = fig.pos.x;
+        let ypos : number = fig.pos.y;
+
+        for(let i=0; i < 3; ++i){
+            fig.dim += 5;
+            fig.display();
+        }
+        
+        var shape : SelectedShape;
+        var cache;
+        // var color;
+        
+        switch(type) {
+            case("CIRCLE"):
+                cache = this.circleCache;
+                shape = SelectedShape.Circle;
+                // color = getCircleColor();
+                break;
+            case("RECT"):
+                xpos += dim/2;
+                ypos += dim/2;
+
+                cache = this.squareCache;
+                shape = SelectedShape.Rectangle;
+                // color = getRectColor();
+                break;
+            case("POLY"):
+                cache = this.triangleCache;
+                shape = SelectedShape.Triangle;
+                // color = getTriangleColor();
+                break;
+        }
+
+        if(dim > 25) { 
+            if(shape && sketchData.figs.length + 5 <= this.NUM_FIGURES) {
+                let newFig : AnimatedFigure;
+                let numNewFigs = dim/10 > 5 ? 5 : Math.floor(dim/10);
+                numNewFigs = Math.min(numNewFigs, cache.length);
+                for(let i = 0; i < numNewFigs; ++i) {
+                    newFig = cache.shift();
+                    newFig.pos.x = xpos + ((Math.random() - 0.5) * 2)*dim/2;
+                    newFig.pos.y = ypos + ((Math.random() - 0.5) * 2)*dim/2;
+                    newFig.dim = dim * (0.3 + Math.random() * 0.3);
+
+                    sketchData.figs.push(newFig);
+                }
+            }
         }
+        let recycleFig = sketchData.figs.splice(index, 1)[0]
+        // recycleFig.color = 
+        cache.push(recycleFig);
+        fig.pop.play();
     }
+    
+    static burst(fig, sketchData, p) {
+        let index = sketchData.figs.indexOf(fig);
+
+        let type : string = fig.getShapeDescriptor().type;
+        let dim : number = fig.dim;
+        let xpos : number = fig.pos.x;
+        let ypos : number = fig.pos.y;
+
+        for(let i=0; i < 3; ++i){
+            fig.dim += 5;
+            fig.display();
+        }
```

Comment:
this loop does the same thing as `fig.dim += 15; fig.display();`. I can explain more or you can read into it, but the state of the canvas is determined by the all of the side effects after calling draw(). If you want the size to grow for a couple of frames you'll have to do something different. For example, keep a variable `fig.to_burst = -1`, when its popped, `fig.to_burst = 0`, then have an if statement instead of this for loop that says something like,
`if (to_burst >= 0 && to_burst < 3) {  fig.dim += 5;   fig.to_burst++;  fig.display();  }`. Let me know if this doesn't help.